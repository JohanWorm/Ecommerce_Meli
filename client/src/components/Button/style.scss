@import 'src/assets/styles/variables.scss';

.btn {
    outline         : none;
    border          : none;
    width           : 100%;
    cursor          : pointer;
    text-align      : center;
    background-color: $color__button;
    border-radius   : $radius;
    height          : 40px;
    color           : $color__white;
    font-size       : 16px;
}