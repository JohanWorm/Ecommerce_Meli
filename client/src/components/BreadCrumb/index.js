import React from 'react';
import './style.scss';

const BreadCrumb = (props) => {
    return (
        <li className="bread__crumb cursor-pointer">
            {props.name}
        </li>
    );
}

export { BreadCrumb };