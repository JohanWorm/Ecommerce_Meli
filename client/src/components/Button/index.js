import React from 'react';
import './style.scss';

const Button = (props) => {

    return (
        <button
            className='btn'
            type={props.type}>
            {props.title}
        </button>
    );
}

export { Button };