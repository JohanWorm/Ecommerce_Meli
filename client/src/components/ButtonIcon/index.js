import React from 'react';
import './style.scss';

const ButtonIcon = (props) => {

  return (
    <button
      className='btn__icon'
      color={props.color}
      type={props.type}>
      <img src={props.src}
        alt='Icono' />
    </button>
  );
}

export { ButtonIcon };