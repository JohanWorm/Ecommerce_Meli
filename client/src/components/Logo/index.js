import React from 'react';
import { useHistory } from 'react-router-dom';
import logo from 'assets/images/Logo_ML.png';
import logo2x from 'assets/images/Logo_ML@2x.png.png.png';
import 'src/assets/styles/global.scss';
import './style.scss';

const Logo = () => {
    
    let history = useHistory();

    const goToHome = () => {
        history.push(`/`);
    };

    return (
        <img className="logo h-100" src={logo2x} alt="Logo" onClick={goToHome} />
    );
}

export { Logo };