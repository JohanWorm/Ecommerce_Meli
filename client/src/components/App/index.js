import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import routes from 'src/routes'
import { Header } from '../Header';
import './style.scss';

const App = () => {

  const history = createBrowserHistory();

  return (
    <main className='app'>
      <Router history={history}>
        <Header />
        <Switch>
          {routes.map((route, i) => <Route key={i} {...route} />)}
        </Switch>
      </Router>
    </main>
  );
}

export { App };
