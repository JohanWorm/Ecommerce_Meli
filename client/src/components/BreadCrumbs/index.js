import React from 'react';
import { BreadCrumb } from 'src/components/BreadCrumb';
import './style.scss';

const BreadCrumbs = (props) => {
  return (
    <ul className='bread__crumbs'>
      {props.categories.map((category, key) => <BreadCrumb key={key} name={category}></BreadCrumb>)}
    </ul>
  );
}

export { BreadCrumbs };