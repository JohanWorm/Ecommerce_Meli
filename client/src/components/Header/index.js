import React from 'react';
import { Logo } from 'src/components/Logo';
import { SearchBox } from 'src/components/SearchBox';
import './style.scss';

const Header = () => {
    return (
        <header className="header">
            <nav className='header__nav container'>
                <Logo />
                <SearchBox />
            </nav>
        </header>
    );
}

export { Header };