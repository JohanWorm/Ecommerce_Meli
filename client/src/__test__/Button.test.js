import React from 'react';
import { shallow } from 'enzyme';
import { Button } from '../components/Button';
import { create } from 'react-test-renderer';

describe('<Button />', () => {
  test('Should render without problems', () => {
    shallow(<Button />);
  })
});

describe('<Button /> Snapshot', () => {
  test('Should render the same interface as the Snapshot', () => {
    const button = create(<Button />);
    expect(button.toJSON()).toMatchSnapshot();
  })
});