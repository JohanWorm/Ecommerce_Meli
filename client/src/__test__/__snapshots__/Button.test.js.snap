// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshot Comprobar interfaz <Button /> 1`] = `
<button
  className="btn"
/>
`;
